### ISSUE TITLE
Testing Turbolinks XHR redirects not working in Rails 6.0.0

### ISSUE THREAD

**Issue by dwightwatson** [rails/rails#36989]
### Steps to reproduce
I think I have identified an issue introduced in Rails 6.0.0.rc2 that impacts testing with Turbolinks. It appears to be related to the new config option `action_dispatch.return_only_media_type_on_content_type = true`.

After upgrading my app from Rails 5.2.3 to 6.0.0 and calling `config.load_defaults 6.0` my tests fail when making assertions about XHR redirects.

```rb
describe 'POST #create' do
  it 'redirects to listings#show' do   
    listing = create(:listing)
    
    post enquiries_path(listing), params: { enquiry: attributes_for(:enquiry) }, xhr: true
    
    expect(response).to redirect_to listing_path(listing)
  end
end
```

This test ran correctly on 5.2.3 and Rails 6.0.0.rc1, but now fails.

```
Failure/Error: expect(response).to redirect_to listing_path(listing)
     
  Expected response to be a <3XX: redirect>, but was a <200: OK>
  Response body: Turbolinks.clearCache()
  Turbolinks.visit("http://www.example.com/listings/1", {"action":"replace"})
```

When I explicitly adjust the `action_dispatch.return_only_media_type_on_content_type = false` (the "old" setting) the tests pass again, albeit with a heap of deprecation warnings.

```
DEPRECATION WARNING: Rails 6.1 will return Content-Type header without modification. If you want just the MIME type, please use `#media_type` instead. (called from block (4 levels) in <top (required)> at /Users/dwight/Sites/tight-app/spec/requests/listings/enquiries_spec.rb:184)
```
  
I'm not entirely sure if this is an issue with Rails, its Turbolinks integration or perhaps something to do with RSpec. However, I would have expected testing redirects on XHR requests would still work out of the box.

### Expected behavior
Tests that make an assertion about a redirect with an XHR/Turbolinks request will continue to pass in Rails 6.0.0 as they did in Rails 5.2.3.

### Actual behavior
These tests do not pass, and instead fail: `Expected response to be a <3XX: redirect>, but was a <200: OK>`.

### System configuration
**Rails version**: 6.0.0

**Ruby version**: 2.3.2

**Comment by georgeclaghorn** [comment]
This was fixed in turbolinks/turbolinks-rails@8da1f63. We need to cut a new release of that gem, but for now you can bundle it from GitHub.

/cc @rafaelfranca

**Comment by javan** [comment]
@georgeclaghorn, was this fixed with the release of turbolinks-rails v5.2.1?

**Comment by georgeclaghorn** [comment]
Yep.
